Tests/DatadogTests/Datadog/RUM/RUMVitals/VitalRefreshRateReaderTests.swift

```diff
@@ -41,7 +41,7 @@ class VitalRefreshRateReaderTests: XCTestCase {
                 }
             )
 
-            wait(for: [completion], timeout: 10)
+            wait(for: [completion], timeout: 20)
```

Comment:
When looking into CI app data for this test, I'm not sure if bumping the timeout will have any effect. In average, this test is passing with `~1s` duration, so the problem is clearly not "running close to the timeout":

<img width="1436" alt="Screenshot 2022-04-08 at 10 15 32" src="https://user-images.githubusercontent.com/2358722/162395031-b95fc17a-0db4-4f84-80b1-872d6a3dd384.png">

Also, the timeout is responsible for 9 of 11 failures during last month:

<img width="1433" alt="Screenshot 2022-04-08 at 10 41 26" src="https://user-images.githubusercontent.com/2358722/162399315-3990386d-955f-4792-bf60-e092610be7b6.png">

The other two failures executed in `1s` and `2s` with recording unexpected measurement.

---

This leads me to the conclusion that we have 2 separate problems in:
```swift
record(lowOverheadMeasure, mainThreadWork: lowOverhead)
record(highOverheadMeasure, mainThreadWork: highOverhead)
```
1. Sometimes recording measurement takes 10x more than usual, so exceeds the timeout before fulfilling `until: { measure.currentValue.sampleCount == targetSamplesCount }` condition.
2. Sometimes it completes as expected, but recorded measurement is wrong in `XCTAssertGreaterThan(expectedHighFPS, expectedLowFPS)`.

I think that we need to answer following when moving forward:
* **What does it mean when the timeout is exceeded, but the condition is not met?** 

Maybe the `measure.currentValue.sampleCount` jumps above `targetSamplesCount` for some reason while the condition is `==`? 💡 I think it would be very explaining if we report the `measure.currentValue.sampleCount` as part of the failure message, e.g.:
```
Exceeded timeout of 10 seconds, with unfulfilled expectations: "Complete measurement" and receiving 20 out of 30 expected samples
```
This can be done if we use `XCTWaiter()` and `XCTFail` APIs [like we do here](https://github.com/DataDog/dd-sdk-ios/blob/34748605ea313798674b062d98e8cdf4d9394b72/Tests/DatadogTests/Datadog/Mocks/ServerMock.swift#L200).

* **Why does it measure `59.99(9)` in `expectedLowFPS`?**

This means we're receiving almost max FPS measurement although we clearly block the main thread with `Thread.sleep(forTimeInterval: 0.02)` 🤨. IMO, to make it cleaner we could record the number of times the `highOverhead` block was called. If it turns out that we're calling it, but still reporting `~60FPS`, that might indicate a problem with the implementation and not test flakiness.

WDYT @maxep ?

Reply:
Indeed, it makes total sense! Thanks for collecting the data 👍 
I will revert and add more details to test failures. And I will see if we can fall into `measure.currentValue.sampleCount` greater than `targetSamplesCount`